### DISCUSSION TITLE
什么时候需要在页面顶部引入 react ?

### DISCUSSION THREAD

**Discussion by vaynevayne** [umijs/dumi#1673] [Q&A]
业务项目报错
<img width="547" alt="image" src="https://github.com/umijs/dumi/assets/124666577/fea922b7-bfce-4379-9c87-6181a5624d02">

以前需要在顶部引入react,  后来不是不需要了吗?  怎么知道何时需要, 何时不需要啊? 
.md 文件里使用组件是正常的, 业务项目引入就各种报错

<img width="770" alt="image" src="https://github.com/umijs/dumi/assets/124666577/12f4ac95-cc0d-4a7f-8168-ca0ddaa5df45">

**Comment by mortalYoung** [comment]
参考一下：https://zh-hans.legacy.reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#whats-a-jsx-transform

**Reply by vaynevayne** [reply]
多谢, 按这个说法react17就支持了啊, 我这个都react18了, 那为啥还需要引用呢

**Reply by mortalYoung** [reply]
除了 react 的版本有要求以外，编译器也需要兼容。文章里有说，如果是 babel 的话该如何兼容等等。

**Reply by vaynevayne** [reply]
多谢, 我还以为开箱即用呢

**Comment by leegsen7** [comment]
如果是自建配置可以看下这个：https://zh-hans.legacy.reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#manual-babel-setup
